babel/src/test/java/org/apache/calcite/test/BabelTest.java

```diff
@@ -239,6 +239,47 @@ private void checkInfixCast(Statement statement, String typeName, int sqlType)
         "EXPR$0=10\n");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5087">[CALCITE-5087]
+   * Support bitwise operators</a>. */
+  @Test void testBitwiseOperator() {
+    final SqlValidatorFixture f = Fixtures.forValidator()
+        .withParserConfig(c -> c.withParserFactory(SqlBabelParserImpl.FACTORY))
+        .withOperatorTable(operatorTableFor(SqlLibrary.MYSQL));
+
+    f.withSql("select 1 | 1")
```

Comment:
As commented below, BIGINT seems like the wrong type.
I would also add some tests that require implicit casts, like operations between int and tinyint values, to make sure that the type coercion rules work correctly.